### PR TITLE
add gradle

### PR DESCRIPTION
**Pull request by klugem**
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by klugem** [comment]
required to build #9652

File: recipes/gradle/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    - openjdk >=11
+
+test:
+  commands:
+    - gradle -v
+
+about:
+  home: https://gradle.org
+  license: Apache 2.0
+  license_file: LICENSE.txt
```
Comment (chrisburr) [inline recipes/gradle/meta.yaml]:
```suggestion
  license_file: LICENSE
```

File: recipes/gradle/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  run:
+    - openjdk >=11
+
+test:
+  commands:
+    - gradle -v
+
+about:
+  home: https://gradle.org
+  license: Apache 2.0
```
Comment (chrisburr) [inline recipes/gradle/meta.yaml]:
```suggestion
  license: Apache-2.0
```

File: recipes/gradle/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+{% set name = "gradle" %}
+{% set version = "5.3.1" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/gradle/gradle/archive/v5.3.1.tar.gz
```
Comment (chrisburr) [inline recipes/gradle/meta.yaml]:
```suggestion
  url: https://github.com/gradle/gradle/archive/v{{ version }}.tar.gz
```

File: recipes/gradle/LICENSE.txt [hunk 4]
```diff
@@ -0,0 +1,420 @@
+
```
Comment (chrisburr) [inline recipes/gradle/LICENSE.txt]: No need to include the licence in the recipe directory as it's already in the tarball

Comment (klugem) [reply recipes/gradle/LICENSE.txt]: I removed the file.

File: recipes/gradle/meta.yaml (hunk 2 above)
Comment (klugem) [reply recipes/gradle/meta.yaml]: done :)

File: recipes/gradle/meta.yaml (hunk 3 above)
Comment (klugem) [reply recipes/gradle/meta.yaml]: dito 👍

File: recipes/gradle/meta.yaml (hunk 1 above)
Comment (klugem) [reply recipes/gradle/meta.yaml]: > license_file entry is missing, but is expected.
Should I leave the "license_file: LICENSE" entry even if the file is not included in the recipe ?

Comment (chrisburr) [reply recipes/gradle/meta.yaml]: Yes, it’ll get it from the tarball, which is generally preferred to ensure it’s kept up to date.

File: recipes/gradle/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+{% set version = "5.3.1" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/gradle/gradle/archive/v{{ version }}.tar.gz
+  sha256: f5bc12b30a44ffe79295c5a9a7ac6b1cfa52e99226cb40e6d18a05af91c73ea4
+  fn: gradle-{{ version }}.tar.gz
```
Comment (dbast) [inline recipes/gradle/meta.yaml]:
```suggestion
```

fn is not require anymore.

File: recipes/gradle/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+build:
+  number: 0
+  skip: true  # [osx or win]
```
Comment (dbast) [inline recipes/gradle/meta.yaml]: why the skip of osx and win?

File: recipes/gradle/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+requirements:
+  host:
+    - coreutils >=8
+    - openjdk >=11
+  run:
+    - openjdk >=11
```
Comment (dbast) [inline recipes/gradle/meta.yaml]: can we remove this from run to let user decide if he wants to use the system java or install it via conda?

Comment (dbast) [reply recipes/gradle/meta.yaml]: It can be added to `test:` to make the tests work.

File: recipes/gradle/meta.yaml (hunk 6 above)
Comment (klugem) [reply recipes/gradle/meta.yaml]: no clue why windows is not working but I enabled osx

**Comment by dbast** [comment]
Seems to need a retrigger as it fails after feedstock conversion https://github.com/conda-forge/gradle-feedstock/commits/master

**Comment by klugem** [comment]
> Seems to need a retrigger as it fails after feedstock conversion https://github.com/conda-forge/gradle-feedstock/commits/master

There seems to be a difference in the build steps between feedstock and staged-r. build causing that error..

I added the same code as in the feedstock here #9666 and it works...

**Comment by klugem** [comment]
**CI** environment variable caused the problem.

**Comment by dbast** [comment]
Thanks for bringing gradle to conda-forge and figuring this out / fixing this!

**Pull request merged**